features/step_definitions/delivery_methods_steps.rb

```diff
@@ -36,7 +36,7 @@
 end
 
 Then(/^the customer should be notified by email$/) do
-  last_job = ActiveJob::Base.queue_adapter.enqueued_jobs.last
+  last_job = ActiveJob::Base.queue_adapter.enqueued_jobs.reverse.detect { |job| job[:job] == NotifyViaEmail }
```

Comment:
It wasn't about the approach for retrieving the job, it was more the fact you're digging way down into jobs from the perspective of a user-facing test. I'd favour checking on something in the presence of the page the *user* sees as confirmation.

You could then check a job is enqueued from the perspective of a unit test around this behaviour.